csrc/ir/internal_nodes.h

```diff
@@ -568,6 +568,79 @@ class TORCH_CUDA_CU_API RNGOp : public Expr {
   }
 };
 
+//! FunctionalRNGOp is just like RNGOp but takes explicit Vals for the Philox
+//! seed and offset so it is deterministic.
+class TORCH_CUDA_CU_API FunctionalRNGOp : public Expr {
```

Comment:
Do we need a new expr type? How is this behaving differently as the non-functional one? Using the same expr type is more aligned with the change I am planning to make as described in https://github.com/NVIDIA/Fuser/issues/535#issuecomment-1607849936, but if there are some blockers, I am also OK with adding this new expr type, but likely I will remove it in the future.

Reply:
I started by trying to reuse `RNGOp`, but the seed and offset are additional inputs, and it was cumbersome to test whether the seed was specified each time it was used. But I think I now after going through the codegen part, I understand your approach. Creating that `GetPhiloxSeedAndOffsetFromATen` is much better since we could make use of that op for other expressions potentially. In this case, when determining whether to pass the additional `philox_args` argument, we would look in `summary_` for presence of that one `Expr`.

For this PR I could try making that change, but leaving the `kernel_inputs` part to you. Do you think that would make sense?

Reply:
That absolutely makes sense.

Reply:
I just pushed a change that removes `FunctionalRNGOp`. Now `RNGOp` can take optional `Val*` arguments `philox_seed` and `philox_offset`. I also provide `RNGOp::getRNGSeedVal()` and `RNGOp::getRNGOffsetVal()` which return the `Val*`s that were provided (or `nullptr)` and `isDeterministic()` to test whether those are present. In codegen, I just check those pointers and use them instead of referring to `philox_args` if non-null.